plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -317,6 +317,8 @@
 import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
 
+// staging directory is shared mutable state
+@Test(singleThreaded = true)
```

Comment:
Will that enforce test from both subclasses to be run mutually exclusive?
Or we do not care?

Reply:
Will not. I think it matters in this case. Can we configure each query runner with a random staging directory?

Reply:
> Will that enforce tests from ~both~ all subclasses to be run mutually exclusive?

it won't. i mean it for doc purposes.


> Can we configure each query runner with a random staging directory?

`temporaryStagingDirectory` is set to a new directory in `@BeforeClass`